### PR TITLE
fix tests after ansible changes

### PR DESCRIPTION
**Pull request by evgeni**
pip doesn't like multiple mentions of the same thing

### CODE REVIEW

**Comment by evgeni** [comment]
bleh, the log was full of

```
2021-09-21T15:02:14.3382950Z [0;35m[DEPRECATION WARNING]: Ansible will require Python 3.8 or newer on the [0m
2021-09-21T15:02:14.3384124Z [0;35mcontroller starting with Ansible 2.12. Current version: 3.6.15 (default, Sep  6[0m
2021-09-21T15:02:14.3385262Z [0;35m 2021, 07:07:23) [GCC 9.3.0]. This feature will be removed from ansible-core in[0m
2021-09-21T15:02:14.3386303Z [0;35m version 2.12. Deprecation warnings can be disabled by setting [0m
2021-09-21T15:02:14.3387317Z [0;35mdeprecation_warnings=False in ansible.cfg.[0m
```

**Comment by evgeni** [comment]
wow, this was harder than it should have been, but :green_apple:

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -931,7 +931,7 @@ def auto_lookup_entities(self):
     def auto_lookup_nested_entities(self):
         for key, entity_spec in self.foreman_spec.items():
             if entity_spec.get('type') in {'nested_list'}:
-                for nested_key, nested_spec in self.foreman_spec[key]['foreman_spec'].items():
+                for nested_key, nested_spec in entity_spec['foreman_spec'].items():
```
Comment (mdellweg) [inline plugins/module_utils/foreman_helper.py:934]: This is a no change, right?

File: plugins/callback/foreman.py [hunk 2]
```diff
@@ -242,7 +242,7 @@ def send_facts(self):
             facts = {
                 "name": host,
                 "facts": {
-                    "ansible_facts": self.facts[host],
+                    "ansible_facts": facts,
```
Comment (mdellweg) [inline plugins/callback/foreman.py:245]: so is this...

File: requirements-dev.txt [hunk 3]
```diff
@@ -14,4 +14,4 @@ cryptography<3.1; python_version < '3.6'
 -r requirements-lint.txt
 -r requirements.txt
 pylint==2.6.0; python_version >= '3.6'
--r https://github.com/ansible/ansible/raw/devel/test/lib/ansible_test/_data/requirements/sanity.validate-modules.txt
+voluptuous==0.12.1 # from https://github.com/ansible/ansible/raw/devel/test/lib/ansible_test/_data/requirements/sanity.validate-modules.txt
```
Comment (mdellweg) [inline requirements-dev.txt:17]: So this must be the meat and potato part?

**Review by mdellweg — APPROVED** [review]

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (evgeni) [reply plugins/module_utils/foreman_helper.py:934]: Correct, just something pylint spotted as being unnecessary

File: plugins/callback/foreman.py (hunk 2 above)
Comment (evgeni) [reply plugins/callback/foreman.py:245]: Same, yeah

**Pull request merged**